k8s/docs/change-control-process.md

```diff
@@ -0,0 +1,106 @@
+
+For example, there is no complete analog for local development and testing; only
+incomplete approaches with various tradeoffs.
+
+ - You could stand up your own local kubernetes cluster using something like
+   [minikube](https://minikube.sigs.k8s.io/docs/) and iterate on that, but there
+   are a number of facilities that are hard or impossible to reproduce locally,
+   such as DNS and certificate management.  Also, many of the file contents in
+   this repo contain portions that are specific to the production cluster, and
+   would likely not work for a local development cluster without modification.
+
+ - You could deploy resources that you are iterating on directly to the
+   production cluster, but this also has drawbacks.  Also, depending on what
+   changes you'd like to experiment with, iteration on the live production
+   system may not be possible due to the change control software preventing or
+   automatically undoing the changes.
+
+#### 1 - Staging, Patch Files, and the main Branch
+
+The spack infrastructure repository has a number of key branches that serve
+different purposes.  First among these is the `main` branch.  This branch
+contains the latest head of development and is the branch that most pull
+requests will be made against.  The overall change control process starts with
+these pull requests.
+
+Among the files in the repository are a number of special config maps
+representing "staged resources" - resources that are deployed for staging
+purposes.  Such resources are automatically deployed from the `main` branch and
+updated when new commits are merged to the `main` branch.
+
+#### 2 - Production and Maintenance Windows
+
+The contents of the `production` branch represent the resources that are
+deployed and managed for production services.  The `production` branch typically
+lags behind the `main` branch and is periodically brought up-to-date with the
+`main` branch.  In general, these pushes to `production` are done during a
+maintenance window - a predetermined window of time within which production
+services may become unavailable.  While there are no formal service level
+agreements for the production services, the regularly-scheduled maintenance
+window affords us the option to perform service-disrupting maintanenace work
+within a specific period of time while providing service availability on a
+best-effort basis outside of that period.  Currently, the maintenance windows
+occur on the first and third Friday of every month, starting on midnight,
+eastern time, on that Friday and ending on midnight, eastern time, the next day.
+
+#### 3 - The flux Branch
+
+The `flux` branch is a branch that is automatically managed by middleware
+running on the cluster.  Its contents represent the exact resources that are
+provisioned on the cluster at any given time and are generated by combing the
+contents of the `main` and `production` branches for staged and production
+resources, respectively.
```

Comment:
Maybe add a note like "You would generally not open a pull request against or interact with this branch" (is that is the case). Also flux might get confused with https://github.com/flux-framework/flux-core